### PR TITLE
PDE-5309 feat(core): add support for "buffered creates"

### PR DESCRIPTION
**Pull request by eliangcs**
Introducing "Buffered Create Actions"...

A Buffered Create allows you to create objects in bulk with a single or fewer API request(s). This is useful when you want to reduce the number of requests made to your server. When enabled, Zapier holds the data until the buffer reaches a size limit or a certain time has passed, then sends the buffered data using the `performBuffer` function you define.

To implement a Buffered Create, you define a [`buffer`](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#bufferconfigschema) configuration object and a `performBuffer` function in the `operation` object. In the `buffer` config object, you specify how you want to group the buffered data using the `groupedBy` setting and the maximum number of items to buffer using the `limit` setting.

Here's an example:

```js
const performBuffer = async (z, bufferedBundle) => {
  // Send a request to create tasks in bulk
  const response = await z.request({
    method: 'POST',
    url: 'https://httpbin.zapier-tooling.com/post',
    body: {
      items: bufferedBundle.buffer,
    },
  });

  const items = response.data.json.items;
  const result = {};

  for (const item of items) {
    const idempotencyId = item.meta.id;
    const itemProps = item.inputData;
    result[idempotencyId] = {
      outputData: {
        id: Date.now(),
        ...itemProps,
      },
    };
  }

  return result;
};

module.exports = {
  key: 'task_buffered',
  noun: 'Task (Buffered)',

  display: {
    label: 'Create Task (Buffered)',
    description: 'Creates a task.',
  },

  operation: {
    performBuffer,
    buffer: {
      groupedBy: ['project_id'],
      limit: 100,
    },

    inputFields: [
      { key: 'project_id', required: true },
      { key: 'title', required: true },
      { key: 'content' },
    ],

    sample: {
      id: 1,
      name: 'Test',
    },

    outputFields: [],
  },
};
```

### CODE REVIEW

**Comment by standielpls** [comment]
I reviewed bc it was referenced in the backend MR, but noticed this was still in draft, so did a quick pass through, lmk when this is ready!

File: packages/schema/test/functional-constraints/bufferedWriteConstraints.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      display: {
+        label: 'Create Foo',
+        description: 'Creates a...',
+      },
+      operation,
+    },
+  },
+};
+
+describe('bufferedWriteConstraints', () => {
```
[the remark this answers is not in the document]
Comment (eliangcs) [reply packages/schema/test/functional-constraints/bufferedWriteConstraints.js]: Sorry I'm not sure what you mean. Can you clarify?

File: packages/schema/lib/schemas/BasicCreateActionOperationSchema.js [hunk 2]
```diff
@@ -20,7 +23,45 @@ BasicCreateActionOperationSchema.properties.shouldLock = {
...
+    required: {
+      type: 'replace', // replace or append
+      value: 'no (with exceptions, see description)',
+    },
+  },
+};
+
+BasicCreateActionOperationSchema.properties.buffer = {
+  description:
+    'Zapier uses this configuration for writing in bulk with `performBuffer`.',
```
[the remark this answers is not in the document]
Comment (eliangcs) [reply packages/schema/lib/schemas/BasicCreateActionOperationSchema.js]: Good call! I've replaced "write" with "create" in several places I found.

File: packages/schema/lib/schemas/BufferConfigSchema.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  properties: {
+    groupedBy: {
+      description:
+        'The list of keys of input fields to group bulk-write with. The actual user data provided for the fields will be used during execution. Note that a required input field should be referenced to get user data always.',
+      type: 'array',
+      minItems: 1,
+    },
+    limit: {
+      description:
+        "The maximum number of items to call `performBuffer` with. **Note** that it is capped by the platform to prevent exceeding the [AWS Lambda's request/response payload size quota of 6 MB](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html#function-configuration-deployment-and-execution). Also, the execution is time-bound; we recommend reducing it upon consistent timeout.",
```
[the remark this answers is not in the document]
Comment (eliangcs) [reply packages/schema/lib/schemas/BufferConfigSchema.js:19]: It was written before we added supported for large response payloads. I think for response payloads, it's no longer relevant. But for _request_ payloads, it's still relevant, so I think we should keep it.

File: packages/schema/test/functional-constraints/bufferedWriteConstraints.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (standielpls) [reply packages/schema/test/functional-constraints/bufferedWriteConstraints.js]: Ignore me, I thought I only saw test cases for all the various error paths, but didn't see a happy success path, but I see it now.

**Review by standielpls — APPROVED** [review]
Changes look great, thank you. 
I wasn't able to test this on an integration (found issues upstream, likely an environment thing) but it looks like you tested at length in the demo, so I'm happy with that.

**Comment by eliangcs** [comment]
@standielpls thanks for the review! I added some text mentioning the feature is internal-only. Can you re-review?

File: packages/core/types/zapier.generated.d.ts [hunk 4]
```diff
@@ -1385,6 +1415,21 @@ export interface BasicCreateActionOperation {
    * concurrency)?
    */
   shouldLock?: boolean;
+
+  /**
+   * Currently an **internaly-only** feature. Zapier uses this
```
Comment (standielpls) [inline packages/core/types/zapier.generated.d.ts:1420]: Found a typo here for internal

File: packages/schema/lib/schemas/BasicCreateActionOperationSchema.js [hunk 5]
```diff
@@ -20,7 +23,45 @@ BasicCreateActionOperationSchema.properties.shouldLock = {
...
+    required: {
+      type: 'replace', // replace or append
+      value: 'no (with exceptions, see description)',
+    },
+  },
+};
+
+BasicCreateActionOperationSchema.properties.buffer = {
+  description:
+    'Currently an **internaly-only** feature. Zapier uses this configuration for creating objects in bulk with `performBuffer`.',
```
Comment (standielpls) [inline packages/schema/lib/schemas/BasicCreateActionOperationSchema.js:40]: internal typo

**Review by standielpls — APPROVED** [review]
I'll approve to unblock, just found a couple typos

**Review by szchenghuang — APPROVED** [review]

**Pull request merged**